src/Action.hs

```diff
@@ -67,8 +70,12 @@ data UserState
 
 makeLenses ''UserState
 
+-- TODO: here @ActionPersist Persist@ is hardwired, so either
+-- parameterize ActionM by the persistance implementation
+-- or remove ActionPersist from ActionM and add by hand elsewhere
+-- (messy and lots of places need to be touched).
 class ( ActionLog m
-      , ActionPersist m
+      , ActionPersist Persist m
```

Comment:
@andorp, @fisx: this is still a major abstraction leak. As soon as @fisx generally approves of the effort (earlier, we would risk wasting time), could you have a look here and suggest something? I'd lean towards parameterizing `ActionM` but this is the most important monad ever, so probably I have no clue what I'm proposing.